presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -124,6 +126,98 @@ public void testNoPathFilter()
         assertEquals(drain(hiveSplitSource).size(), 2);
     }
 
+    @Test
+    public void testCsvWithoutHeaderAndFooter()
+            throws Exception
+    {
+        Table table = table(
+                ImmutableList.of(),
+                Optional.empty(),
+                ImmutableMap.of(),
+                StorageFormat.fromHiveStorageFormat(CSV));
+
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = backgroundHiveSplitLoader(
+                ImmutableList.of(locatedFileStatus(new Path(SAMPLE_PATH), new DataSize(2.0, GIGABYTE).toBytes())),
+                TupleDomain.all(),
+                Optional.empty(),
+                table,
+                Optional.empty());
+
+        HiveSplitSource hiveSplitSource = hiveSplitSource(backgroundHiveSplitLoader);
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+
+        assertEquals(drainSplits(hiveSplitSource).size(), 33);
```

Comment:
❤️ 

Maybe  we can now port tests that were using ./src/main/resources/sql-tests/datasets/table_with_header_and_footer.data-generator?